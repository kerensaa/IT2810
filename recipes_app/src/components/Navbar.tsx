import React from 'react'
import '../styling/Navbar.css'

function Navbar() {
    return (
      <div className='navbar-container'>
        <nav className='nav'>
        <div className='left-content'>
            <h2><a href="/" className='link'>Recipes</a></h2>
        </div>
        <div className='right-content'> 
          <button>Search</button>
        </div>
        </nav>
      </div>
    );
  }
export default Navbar