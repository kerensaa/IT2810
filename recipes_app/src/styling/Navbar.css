.nav{
    width:100%;
    background-color: #BB99CD;
    display: grid;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    height: 7rem;
    align-items: center;
    grid-template-columns: 1fr 2fr;
    flex-wrap: wrap;
    z-index: 1000;
}

@media screen and (max-width: 1085px){
  .nav {
    width:100%;
    background-color: #BB99CD;
    display: grid;
    position:absolute;
    top: 0;
    left: 0;
    right: 0;
    height: 7rem;
    align-items: center;
    grid-template-columns: 1fr 2fr;
    flex-wrap: wrap;
    z-index: 1000;
  }
}


.left-content {
    display: flex;
    align-items: center;
    padding-left: 4rem;
    padding-right: 1rem;
    
    /* Typography */
    font-family: Oldenburg;
  }

  .right-content {
    display: flex;
    align-items: center;
    flex-direction: row-reverse;
    padding-right: 2rem;
    gap: 2rem;
    margin-right: 2rem;
    text-decoration: underline;
    text-decoration-color: black;
  }


.link:hover {
    color: black;
}
