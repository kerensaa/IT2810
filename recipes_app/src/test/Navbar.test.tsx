import { render, screen } from '@testing-library/react';
import { LinkProps } from 'react-router-dom';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import Navbar from '../components/Navbar';

vi.mock('react-router-dom', () => ({
  Link: ({ to, children, ...props }: LinkProps) => (
    <a href={typeof to === 'string' ? to : undefined} {...props}>
      {children}
    </a>
  ),
}));

describe('Renders Navbar correctly', async () => {
  beforeEach(() => {
    render(<Navbar />);
  });
  it('Renders NavBar correctly', () => {
    const logo = screen.getByText('Recipes4You');
    expect(logo).not.toBeNull;
  });
});

describe('Snapshot testing of Navbar', async () => {
  it('should perform snapshot test Navbar', async () => {
    const navbar = render(<Navbar />);
    expect(navbar).toMatchSnapshot();
  });
});
