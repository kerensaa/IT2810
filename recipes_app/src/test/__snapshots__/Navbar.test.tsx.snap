// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot testing of Navbar > should perform snapshot test Navbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar-container"
      >
        <nav
          class="nav"
        >
          <div
            class="left-content"
          >
            <h2>
              <a
                href="/"
              >
                Recipes4You
              </a>
            </h2>
          </div>
          <div
            class="right-content"
          >
            <p>
              <a
                href="/MyRatings"
              >
                My Ratings
              </a>
            </p>
            <p>
              <a
                href="/MyFavorites"
              >
                My Favorites
              </a>
            </p>
          </div>
        </nav>
      </div>
    </div>
    <div>
      <div
        class="navbar-container"
      >
        <nav
          class="nav"
        >
          <div
            class="left-content"
          >
            <h2>
              <a
                href="/"
              >
                Recipes4You
              </a>
            </h2>
          </div>
          <div
            class="right-content"
          >
            <p>
              <a
                href="/MyRatings"
              >
                My Ratings
              </a>
            </p>
            <p>
              <a
                href="/MyFavorites"
              >
                My Favorites
              </a>
            </p>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar-container"
    >
      <nav
        class="nav"
      >
        <div
          class="left-content"
        >
          <h2>
            <a
              href="/"
            >
              Recipes4You
            </a>
          </h2>
        </div>
        <div
          class="right-content"
        >
          <p>
            <a
              href="/MyRatings"
            >
              My Ratings
            </a>
          </p>
          <p>
            <a
              href="/MyFavorites"
            >
              My Favorites
            </a>
          </p>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
